Lib/test/test_ast.py

```diff
@@ -1199,6 +1216,247 @@ def test_copy_with_parents(self):
                 )):
                     self.assertEqual(to_tuple(child.parent), to_tuple(node))
 
+    def test_replace_interface(self):
+        for klass in self.iter_ast_classes():
+            with self.subTest(klass=klass):
+                self.assertTrue(hasattr(klass, '__replace__'))
+
+            fields = set(klass._fields)
+            with self.subTest(klass=klass, fields=fields):
+                node = klass(**dict.fromkeys(fields))
+                # forbid positional arguments in replace()
+                self.assertRaises(TypeError, copy.replace, node, 1)
+                self.assertRaises(TypeError, node.__replace__, 1)
+
+    def test_replace_native(self):
+        for klass in self.iter_ast_classes():
+            fields = set(klass._fields)
+            with self.subTest(klass=klass, fields=fields):
+                # use of object() to ensure that '==' and 'is'
+                # behave similarly in ast.compare(node, repl)
+                old_value, new_value = object(), object()
+
+                # check shallow copy
+                node = klass(**dict.fromkeys(fields, old_value))
```

Comment:
Shouldn't this use a different `old_value` for each field?

Reply:
Mmh, technically we don't really care here. It's just to avoid the warning that I did that (mainly to fill up the values that the constructor expects). here, it's mainly to check that `copy.replace(repl)` (when you do nothing) does nothing else. But I'll make them all independent just in case.